Source/Wpf/Prism.Wpf/Regions/RegionNavigationService.cs

```diff
@@ -239,11 +255,12 @@ private void DoNavigate(Uri source, Action<NavigationResult> navigationCallback,
                 navigationContext,
                 navigationCallback,
                 activeViews,
-                currentViewIndex + 1);
+                currentViewIndex + 1,
+                savePrevious);
         }
 
         [SuppressMessage("Microsoft.Design", "CA1031:DoNotCatchGeneralExceptionTypes", Justification = "Exception is marshalled to callback")]
-        private void ExecuteNavigation(NavigationContext navigationContext, object[] activeViews, Action<NavigationResult> navigationCallback)
+        private void ExecuteNavigation(NavigationContext navigationContext, object[] activeViews, Action<NavigationResult> navigationCallback, bool savePrevious)
         {
             try
             {
```

Comment:
Can the PersistInHistory check be done here instead of at line 146? Seems we have all the info we need. This would remove the need to pass the `savePrevious` parameter around.

Reply:
Sure, this can be done. I was starting the analysis early because it might be needed for some scenarios with multiple active views.